scripts/db_migrator.py

```diff
@@ -96,6 +100,108 @@ def migrate_interface_table(self):
+                                  "egress_lossless_pool": { "size": "14024599", "type": "egress", "mode": "dynamic" },
+                                  "egress_lossy_pool": {"size": "5029836", "type": "egress", "mode": "dynamic" } }
+        spc1_t1_default_config = {"ingress_lossless_pool": { "size": "2097100", "type": "ingress", "mode": "dynamic" },
+                                  "ingress_lossy_pool": { "size": "2097100", "type": "ingress", "mode": "dynamic" },
+                                  "egress_lossless_pool": { "size": "14024599", "type": "egress", "mode": "dynamic" },
+                                  "egress_lossy_pool": {"size": "2097100", "type": "egress", "mode": "dynamic" } }
+        spc2_t0_default_config = {"ingress_lossless_pool": { "size": "14983147", "type": "ingress", "mode": "dynamic" },
+                                  "ingress_lossy_pool": { "size": "14983147", "type": "ingress", "mode": "dynamic" },
+                                  "egress_lossless_pool": { "size": "34340822", "type": "egress", "mode": "dynamic" },
+                                  "egress_lossy_pool": {"size": "14983147", "type": "egress", "mode": "dynamic" } }
+        spc2_t1_default_config = {"ingress_lossless_pool": { "size": "9158635", "type": "ingress", "mode": "dynamic" },
+                                  "ingress_lossy_pool": { "size": "9158635", "type": "ingress", "mode": "dynamic" },
+                                  "egress_lossless_pool": { "size": "34340822", "type": "egress", "mode": "dynamic" },
+                                  "egress_lossy_pool": {"size": "9158635", "type": "egress", "mode": "dynamic" } }
+
+        # Get platform info and hwsku from config_db.json
+        buffer_pool_conf = {}
+        config_db_file_path = "/etc/sonic/config_db.json"
+        if os.path.exists(config_db_file_path) and os.path.getsize(config_db_file_path) > 0:
+            try:
+                with open(config_db_file_path) as config_db_file:
+                    config_db = json.load(config_db_file)
+                    if 'DEVICE_METADATA' in config_db.keys():
+                        device_data = config_db['DEVICE_METADATA']
+                        hwsku = device_data['localhost']['hwsku']
+                        platform = device_data['localhost']['platform']
+                    else:
+                        log_error("No DEVICE_METADATA exist in config_db.json file, skip migration")
+                        return False
+                    if 'BUFFER_POOL' in config_db.keys():
+                        buffer_pool_conf = config_db['BUFFER_POOL']
+                    else:
+                        # No buffer pool size configuration exist, no need to migration
+                        log_info("No buffer pool configuration, no need to migrate")
+                        return True
+            except IOError:
+                log_error("failed to open config_db.json file, skip migration")
+                return False
```

Comment:
I think you should remove the if block and only keep the else block. In other words, only migrate the in memory database.

Reply:
Removed. 
In the very beginning, the idea was to get info from config_db.json if it exists, since eventually it will be loaded to the in-memory database, so we can just get data from in-memory DB.

And also added special handling for 3800 platforms, since it has unique buffer pool configuration compare to other Spectrum 2 platform: https://github.com/Azure/sonic-buildimage/pull/4239